packages/client/src/v2-events/features/events/useEvents/procedures/delete.ts

```diff
@@ -27,7 +35,12 @@ function waitUntilEventIsCreated<R>(
 }
 
 utils.event.delete.setMutationDefaults(({ canonicalMutationFn }) => ({
-  retry: true,
+  retry: (_, error) => {
+    if (error.data?.httpStatus === 404 || error.data?.httpStatus === 400) {
```

Comment:
This stops the event deletion from retrying if the endpoint returned with 404 (event most likely already removed) or 400 (it can't be deleted)